mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -727,6 +727,194 @@ public enum MremapFlags : ulong {
 		MREMAP_MAYMOVE = 0x1,
 	}
 
+	[Map]
+	[CLSCompliant (false)]
+	public enum SockType : int {
+		SOCK_STREAM    =  1, // Byte-stream socket
+		SOCK_DGRAM     =  2, // Datagram socket
+		SOCK_RAW       =  3, // Raw protocol interface (linux specific)
+		SOCK_RDM       =  4, // Reliably-delivered messages (linux specific)
+		SOCK_SEQPACKET =  5, // Sequenced-packet socket
+		SOCK_DCCP      =  6, // Datagram Congestion Control Protocol (linux specific)
+		SOCK_PACKET    = 10, // Linux specific
+	}
+
+	[Map][Flags]
+	[CLSCompliant (false)]
+	public enum SockFlags : int {
+		SOCK_CLOEXEC  = 0x80000, /* Atomically set close-on-exec flag for the
+								    new descriptor(s).  */
+		SOCK_NONBLOCK = 0x00800, /* Atomically mark descriptor(s) as
+								    non-blocking.  */
+	}
+
+	[Map]
+	[CLSCompliant (false)]
+	public enum SockProtocol : int {
+		IPPROTO_IP = 0,			/* Dummy protocol for TCP		*/
+		IPPROTO_ICMP = 1,		/* Internet Control Message Protocol	*/
+		IPPROTO_IGMP = 2,		/* Internet Group Management Protocol	*/
+		IPPROTO_IPIP = 4,		/* IPIP tunnels (older KA9Q tunnels use 94) */
+		IPPROTO_TCP = 6,		/* Transmission Control Protocol	*/
+		IPPROTO_EGP = 8,		/* Exterior Gateway Protocol		*/
+		IPPROTO_PUP = 12,		/* PUP protocol				*/
+		IPPROTO_UDP = 17,		/* User Datagram Protocol		*/
+		IPPROTO_IDP = 22,		/* XNS IDP protocol			*/
+		IPPROTO_TP = 29,		/* SO Transport Protocol Class 4	*/
+		IPPROTO_DCCP = 33,		/* Datagram Congestion Control Protocol */
+		IPPROTO_IPV6 = 41,		/* IPv6-in-IPv4 tunnelling		*/
+		IPPROTO_RSVP = 46,		/* RSVP Protocol			*/
+		IPPROTO_GRE = 47,		/* Cisco GRE tunnels (rfc 1701,1702)	*/
+		IPPROTO_ESP = 50,		/* Encapsulation Security Payload protocol */
+		IPPROTO_AH = 51,		/* Authentication Header protocol	*/
+		IPPROTO_MTP = 92,		/* Multicast Transport Protocol		*/
+		IPPROTO_BEETPH = 94,	/* IP option pseudo header for BEET	*/
+		IPPROTO_ENCAP = 98,		/* Encapsulation Header			*/
+		IPPROTO_PIM = 103,		/* Protocol Independent Multicast	*/
+		IPPROTO_COMP = 108,		/* Compression Header Protocol		*/
+		IPPROTO_SCTP = 132,		/* Stream Control Transport Protocol	*/
+		IPPROTO_UDPLITE = 136,	/* UDP-Lite (RFC 3828)			*/
+		IPPROTO_RAW = 255,		/* Raw IP packets			*/
+
+		// Number used by linux (1) clashes with IPPROTO_ICMP
+		SOL_SOCKET = 1024,		/* For setsockopt() / getsockopt(): Options to be accessed at socket level, not protocol level. */
+	}
+
+	[Map]
+	[CLSCompliant (false)]
+	public enum AddrFamily : ushort {
```

Comment:
Yes, I chose ushort becauses Linux uses that. The fact that OS X uses something different is bad, this will break the build because map.c contains this:

    Mono_Posix_FromSockaddr (struct Mono_Posix_Sockaddr *from, struct sockaddr *to)
    ...
    if (Mono_Posix_FromAddrFamily (from->sa_family, &to->sa_family) != 0) {

and `Mono_Posix_FromAddrFamily` expects a pointer to an unsigned int as second parameter.

I'll try to update create-native-map to support that (i.e. to make `Mono_Posix_FromAddrFamily` accept a pointer to a sa_family_t instead).